app/views/user_preferences/index.slim

```diff
@@ -80,66 +80,62 @@ div.content-container
           ' Enable Ultimate Accessibility Mode
           strong #{"[BETA]"}*/
 
-    /*h2 Timezone*/
-    /*p Will adjust all times on the site as you see them into what the respective time is for your timezone.*/
-    /*select name="timezone" id="timezone"
-      option value="-1200" (GMT -12:00) Eniwetok, Kwajalein
-      option value="-1100" (GMT -11:00) Midway Island, Samoa
-      option value="-1000" (GMT -10:00) Hawaii
-      option value="-950" (GMT -9:30) Taiohae
-      option value="-900" (GMT -9:00) Alaska
-      option value="-800" (GMT -8:00) Pacific Time (US &amp; Canada)
-      option value="-700" (GMT -7:00) Mountain Time (US &amp; Canada)
-      option value="-600" (GMT -6:00) Central Time (US &amp; Canada), Mexico City
-      option value="-500" (GMT -5:00) Eastern Time (US &amp; Canada), Bogota, Lima
-      option value="-450" (GMT -4:30) Caracas
-      option value="-400" (GMT -4:00) Atlantic Time (Canada), Caracas, La Paz
-      option value="-350" (GMT -3:30) Newfoundland
-      option value="-300" (GMT -3:00) Brazil, Buenos Aires, Georgetown
-      option value="-200" (GMT -2:00) Mid-Atlantic
-      option value="-100" (GMT -1:00) Azores, Cape Verde Islands
-      option value="0" selected="selected" (GMT) Western Europe Time, London, Lisbon, Casablanca
-      option value="100" (GMT +1:00) Brussels, Copenhagen, Madrid, Paris
-      option value="200" (GMT +2:00) Kaliningrad, South Africa
-      option value="300" (GMT +3:00) Baghdad, Riyadh, Moscow, St. Petersburg
-      option value="350" (GMT +3:30) Tehran
-      option value="400" (GMT +4:00) Abu Dhabi, Muscat, Baku, Tbilisi
-      option value="450" (GMT +4:30) Kabul
-      option value="500" (GMT +5:00) Ekaterinburg, Islamabad, Karachi, Tashkent
-      option value="550" (GMT +5:30) Bombay, Calcutta, Madras, New Delhi
-      option value="575" (GMT +5:45) Kathmandu, Pokhara
-      option value="600" (GMT +6:00) Almaty, Dhaka, Colombo
-      option value="650" (GMT +6:30) Yangon, Mandalay
-      option value="700" (GMT +7:00) Bangkok, Hanoi, Jakarta
-      option value="800" (GMT +8:00) Beijing, Perth, Singapore, Hong Kong
-      option value="875" (GMT +8:45) Eucla
-      option value="900" (GMT +9:00) Tokyo, Seoul, Osaka, Sapporo, Yakutsk
-      option value="950" (GMT +9:30) Adelaide, Darwin
-      option value="1000" (GMT +10:00) Eastern Australia, Guam, Vladivostok
-      option value="1050" (GMT +10:30) Lord Howe Island
-      option value="1100" (GMT +11:00) Magadan, Solomon Islands, New Caledonia
-      option value="1150" (GMT +11:30) Norfolk Island
-      option value="1200" (GMT +12:00) Auckland, Wellington, Fiji, Kamchatka
-      option value="1275" (GMT +12:45) Chatham Islands
-      option value="1300" (GMT +13:00) Apia, Nukualofa
-      option value="1400" (GMT +14:00) Line Islands, Tokelau*/
-    /*span
+    h2 Timezone
```

Comment:
What about https://apidock.com/rails/ActionView/Helpers/FormOptionsHelper/time_zone_options_for_select?

Reply:
So checked it out and although easier and slightly better for low vision users, it is just showing cities, doesn't include day light timezone changes and in my testing of them out, probably 15-20% of them are not working as they should because they are not returning the correct current times from those regions. I don't know how, might be how I'm using it, but it just is.
Rails one liner:
![one line rails TimeZone select](https://user-images.githubusercontent.com/51969207/107529053-7c4bd180-6b88-11eb-9ac2-fa48453cf938.PNG)
Mine:
![my implementation](https://user-images.githubusercontent.com/51969207/107529114-8b328400-6b88-11eb-9dd9-ac7c352e7e3a.PNG)

Been following timeanddate.com for accuracy and insight into the unique timezones that Rails supports that don't conform to those around or do/don't practice daylight savings and marking accordingly. Will only take me another hour to be done with it if you like the dropdown thus far or want me to do something else for this (perhaps make less wordy)?

Reply:
If we don't have a working programmatic source of timezone data for the drop down list, we will not implement this feature. Trying to hand generate the list is going to end up taking a lot of time to do properly to account for all the cases in timezones. Examples from your hand generated list.:
"Indiana (US East)" exists because most of Indiana does not observe Daylight saving time.  
Most of Arizona doesn't do DST (but I don't remember the proper name for this
Neither Bogota nor Lima observe Daylight Savings time, so can't be grouped with US Eastern
And that was just from looking at the US Eastern line, I haven't dug deep on the other times.

Also, from a styling perspective, looking at Windows, Linux and a couple web examples, it is generally also limited to displaying the standard (none DST) offset (UTC-5 for Eastern) not both.  Many things (especially linux based) do use city names as anchors as that can be easier to reference for some people.

Reply:
The Bogota and Lima were left on there from the old list. Fixed it. Indiana time zones are still CST or EST, they just pick the one that is relevant. Yes Arizona doesn't do Daylight savings which is why Rails made them their own endpoint and I made it's own select option (they observe mountain time all year round). We can switch to cities, this just seemed more complete.

I understand you worry about the time suck of doing it by hand, but I did already and got some education on the whole thing. Some of the cities Rails provided didn't work so I used others in the same timezone and those worked. I didn't do one last test yet or lookthrough to see if I mistyped or forgot something and still want to do a little bit more with the ticket, but this is what we're looking at to date (if you are not opposed). Yes timezones change because of global governments but because the full definition of the timezone exists (what the abbreviations are and what it switches to. ex -4 to -5, they will know if it's wrong and they aren't hard to fix should ex. Brazil declare to go back to using daylight savings. If you want me to can it, I can, but it works and I think it's valuable. Certain governments, ex. Myanmar being unstable and potentially changing things shouldn't be a blocker.

![top third](https://user-images.githubusercontent.com/51969207/107545065-d9e81a00-6b98-11eb-8d59-4fa291116a0d.PNG)
![middle third](https://user-images.githubusercontent.com/51969207/107545062-d94f8380-6b98-11eb-8090-289400769084.PNG)
![bottom third](https://user-images.githubusercontent.com/51969207/107545064-d9e81a00-6b98-11eb-968c-453643111be5.PNG)

Reply:
Ok, I stand corrected about Indiana (Indiana US East is still listed separately in most software I have looked at for legacy reasons didn't realize they joined everyone else in 2006).  But, back to Bogota and Lima for example. Bogota is in Columbia Standard Time (which isn't on the list, nor is any other -0500 timezone).  Also, since I just did a test between in the OS and in the container and got a (VERY SLIGHTY but still) different timezone definitions, I am not willing to put out code that uses a hard coded list that might not exactly match what the OS lists as available and cause other issues down the road.  If the list is not generated based off of what is actual available on the system executing the code, it will not be merged.